dashboard/config/foorm/forms/surveys/pd/summer_workshop_post_survey.0.json

```diff
@@ -171,118 +171,601 @@
+        }
+       ]
       },
       {
-       "value": "7",
-       "text": "Strongly Agree"
+       "type": "comment",
+       "name": "question7",
+       "title": "[PLACEHOLDER - CSP Free Response - IF needed]"
       }
      ],
-     "rows": [
-      {
-       "value": "When I need help teaching",
-       "text": "When I need help teaching computer science, I have someone to turn to."
-      },
-      {
-       "value": "I know computer science t",
-       "text": "I know computer science teacher-leaders and role models in my community."
-      },
-      {
-       "value": "I feel like I'm part of a",
-       "text": "I feel like I'm part of a broad computer science education movement."
-      },
-      {
-       "value": "I see my Regional Partner",
-       "text": "I see my Regional Partner as a hub of resources and support for computer science education in my community."
-      }
-     ]
-    },
-    {
-     "type": "html",
-     "name": "question3",
-     "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gt;BLIND SPOTS&lt;&lt;</span></p>\nReplacing Gender/Race blindness/awareness items"
+     "visibleIff": "{workshop_course} = 'CS Principles'",
+     "title": "Teaching CS Principles"
     },
     {
-     "type": "html",
-     "name": "clickTo212",
-     "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gtTEACHER NEEDS (logistics)&lt;&lt;</span></p>\nmore direct ask about teachers concerns/needs about program to help inform implementation (delegate to outreach)"
-    },
-    {
-     "type": "html",
-     "name": "question2",
-     "titleLocation": "hidden",
-     "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gtTEACHER NEEDS (pedagogy, instructional practice)&lt;&lt;</span></p>\nAsking teachers what they want more of, \"what's important to them in terms of followup.\"  a la facilitator followup sessions requests"
-    },
-    {
-     "type": "html",
-     "name": "question1",
-     "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gt;IF CS PRINCIPLES&lt;&lt;</span></p>\nCSP-specifc things we *really* need to know about to inform program\n"
-    },
-    {
-     "type": "html",
-     "name": "question4",
-     "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gt;IF CS DISCOVERIES&lt;&lt;</span></p>\nCSD-specifc things we *really* need to know about to inform program\n"
+     "type": "panel",
+     "name": "csd_panel",
+     "elements": [
+      {
+       "type": "html",
+       "name": "question4",
+       "html": "<p><span style=\"background-color: #ffff00;\">&gt;&gt;IF CS DISCOVERIES&lt;&lt;</span></p>\nCSD-specifc things we *really* need to know about to inform program\n"
+      },
+      {
+          "type": "checkbox",
+          "name": "csd_specific",
+          "title": "Which units of CS Discoveries do you plan to teach in the 2020-2021 school year? (check all that apply)  BAKER NOTE: SEE IF THERE IS WRITE IN 'OTHER' OPTION.",
```

Comment:
there is a write in other option :) You can add the variable "hasOther": true. You can set the option text with "otherText" and the placeholder in the free text with "otherPlaceholder"

Reply:
found it. Added.  Thanks.